extensions/default/src/DicomWebDataSource/index.js

```diff
@@ -252,9 +252,38 @@ function createDicomWebApi(dicomWebConfig, UserAuthenticationService) {
         sortFunction
       );
 
+      /**
+       * naturalizes the dataset, and adds a retrieve bulkdata method
+       * to any values containing BulkDataURI.
+       * @param {*} instance
+       * @returns naturalized dataset, with retrieveBulkData methods
+       */
+      const addRetrieveBulkData = instance => {
+        const naturalized = naturalizeDataset(instance);
+        Object.keys(naturalized).forEach(key => {
+          const value = naturalized[key];
+          if (value && value.BulkDataURI && !value.Value) {
+            // Provide a method to fetch this
+            value.retrieveBulkData = () => {
+              const options = {
+                multipart: false,
+                BulkDataURI: value.BulkDataURI,
+                StudyInstanceUID: naturalized.StudyInstanceUID,
+              };
+              return qidoDicomWebClient.retrieveBulkData(options).then(val => {
+                const ret = val && val[0] || undefined;
```

Comment:
shouldn't we check `val.length` too?

Reply:
No, the API contract is a bit odd, it returns things in an array, or else returns undefined if there isn't anything, so there is
always at least 1 element or else the entire return is null.